frontend/apps/web/app/(mgmt)/[account]/new/connection/aws-s3/AwsS3Form.tsx

```diff
@@ -69,6 +75,65 @@ export default function AwsS3Form() {
       console.error(err);
     }
   }
+  /* we call the underlying useGetConnection API directly since we can't call
+the hook in the useEffect conditionally. This is used to retrieve the values for the clone connection so that we can update the form. 
+*/
+  useEffect(() => {
+    const fetchData = async () => {
+      if (sourceConnId && account?.id) {
+        setIsLoading(true);
+        try {
+          const connData = await GetConnectionCloneValues(
+            account.id,
+            sourceConnId
+          );
+
+          if (connData) {
+            const s3Config = connData.connection?.connectionConfig?.config
+              .value as AwsS3ConnectionConfig;
```

Comment:
No guarantee that the connection id that is provided in the url is actually an S3 connection. So you need to actually check to make sure this is valid here and not type cast it because otherwise it might be undefined.
Use the typeguards to check that it is actually an s3 connection.

Same should be done for the mysql/postgres forms too.

Reply:
updated